xbmc/guilib/GUIFontTTF.cpp

```diff
@@ -455,16 +442,51 @@ void CGUIFontTTF::DrawTextInternal(CGraphicContext& context,
         if (!ch)
           continue;
 
-        textWidth += ch->m_advance;
+        const float nextWidth = textWidth + ch->m_advance;
 
-        if (textWidth > maxPixelWidth)
+        if (nextWidth > maxPixelWidth)
         {
           // Start rendering from the glyph that does not exceed the maximum width
           startPosGlyph = std::distance(itRGlyph, glyphs.crend());
           break;
         }
+        textWidth = nextWidth;
       }
     }
+    else
+    {
+      // Determines the maximum width the text can have within the maximum width
+      if (alignment & XBFONT_TRUNCATED)
+        textWidth = ellipsesWidth;
+
+      for (const Glyph& glyph : glyphs)
+      {
+        Character* ch = GetCharacter(text[glyph.m_glyphInfo.cluster], glyph.m_glyphInfo.codepoint);
+        if (!ch)
```

Comment:
The \t character has a special case in rendering, shouldn't it be the same here? (that would be a defect of the original c/p code)

Reply:
thanks oversight!